mrp_bom_priority/views/view_mrp_bom.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!--
+Copyright (C) 2024 - Today: GRAP (http://www.grap.coop)
+@author: Quentin DUPONT (quentin.dupont@grap.coop)
+License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+-->
+<odoo>
+
+    <!-- ************************************************************ -->
+    <!-- MRP Bill of Materials - Form -->
+    <!-- ************************************************************ -->
+
+    <record id="view_mrp_business_bom_form" model="ir.ui.view">
+        <field name="model">mrp.bom</field>
+        <field name="inherit_id" ref="mrp.mrp_bom_form_view" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='code']" position="after">
+                <field name="priority" widget="priority" />
+            </xpath>
+        </field>
+    </record>
+
+    <!-- ************************************************************ -->
+    <!-- MRP Bill of Materials - Tree -->
+    <!-- ************************************************************ -->
+
+    <record id="view_mrp_bom_tree" model="ir.ui.view">
+        <field name="model">mrp.bom</field>
+        <field name="inherit_id" ref="mrp.mrp_bom_tree_view" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='product_id']" position="after">
+                <field name="priority" widget="priority" />
+            </xpath>
+            <xpath expr="//tree" position="attributes">
+              <attribute name="default_order">priority desc</attribute>
```

Comment:
default_order (or _order), make the item order by the field(s). if there are many items with the same value, the next order is "by id". that is not very user intuitive. See : 

![image](https://github.com/user-attachments/assets/46a45f12-6511-4948-add5-f9e91e660b6f)

In that case, I expect both "extra coulis" boms to be next to it.
something like that ? (not sure, just a proposal).

```suggestion
              <attribute name="default_order">priority desc, product_tmpl_id,code</attribute>
```

Reply:
Yep I think it makes sense !